shared_model/interfaces/iroha_internal/tx_status_factory.hpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_TX_STATUS_FACTORY_HPP
+#define IROHA_TX_STATUS_FACTORY_HPP
+
+#include <memory>
+
+#include "interfaces/transaction_responses/tx_response.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    /**
+     * Factory which creates transaction status response
+     */
+    class TxStatusFactory {
+     public:
+      virtual ~TxStatusFactory() = default;
+
+      using FactoryReturnType = std::unique_ptr<TransactionResponse>;
+
+      using TransactionHashType =
+          const TransactionResponse::TransactionHashType &;
+
+      using ErrorMessageType = const TransactionResponse::ErrorMessageType &;
+
+      // ------------------------| Stateless statuses |-------------------------
+
+      /// Creates stateless failed transaction status
```

Comment:
Not so convenient to do it because there is required to name types in param. Also, documentation for methods will be duplicated. I add type explanations to usings instead.